pkg/reconciler/subscription/subscription.go

```diff
@@ -465,7 +465,17 @@ func (r *Reconciler) patchSubscription(ctx context.Context, namespace string, ch
 	return true, nil
 }
 
-func (r *Reconciler) updateChannelRemoveSubscription(ctx context.Context, channel *eventingduckv1alpha1.Channelable, sub *v1alpha1.Subscription) {
+func (r *Reconciler) updateChannelRemoveSubscription(ctx context.Context, channel *eventingduckv1alpha1.ChannelableCombined, sub *v1alpha1.Subscription) {
+	if channel.Annotations != nil {
+		if channel.Annotations[messaging.SubscribableDuckVersionAnnotation] == "v1beta1" {
+			r.updateChannelRemoveSubscriptionV1Beta1(ctx, channel, sub)
+			return
+		}
+	}
+	r.updateChannelAddSubscriptionV1Alpha1(ctx, channel, sub)
```

Comment:
This is a bug (discovered by @v0id3r). Instead of removing the subscription from v1alpha1 subscriptions, this function adds it, which ends up being a no-op. @v0id3r is working on a PR to fix. Might be worth a backport.